source_code/main_mcu/src/GUI/gui_prompts.c

```diff
@@ -576,6 +576,37 @@ void gui_prompts_display_3line_information_on_screen_and_wait(confirmationText_t
     }
 }
 
+/*! \fn     gui_prompts_get_char_to_display(uint8_t const * current_pin, uint8_t index)
+*   \brief  Get the next PIN value to display or '*' if PIN should be hidden
+*   \param  current_pin         Array containing the pin
+*   \param  index               Index into current_pin array
+*   \return Character to display
+*/
+static cust_char_t gui_prompts_get_char_to_display(uint8_t const * current_pin, uint8_t index)
+{
+    cust_char_t disp_char;
+    uint8_t curr_pin = current_pin[index];
+
+    if ((BOOL)custom_fs_settings_get_device_setting(SETTING_SHOW_PIN_ON_ENTRY) != FALSE)
+    {
+        if (curr_pin >= 0x0A)
+        {
+            disp_char = curr_pin + u'A' - 0x0A;
```

Comment:
maybe use current_pin[index] directly so we don't take the risk of allocating a var that might be read later.

Reply:
I will change it but pretty sure the assembly will look _exactly_ the same. it will be in a register I bet.